internals/daemon/daemon.go

```diff
@@ -58,6 +58,66 @@ var (
 	systemdSdNotify = systemd.SdNotify
 )
 
+// TransportType defines the possible API transport types we support. The
+// type can be extracted from the http.Request using RequestTransportType.
+type TransportType int
+
+// Used with context.WithValue as the key.
+type TransportTypeKey struct{}
+
+const (
+	TransportTypeUnknown TransportType = iota // Must be zero value => 0
+	TransportTypeUnixSocket
+	TransportTypeHTTP
+	TransportTypeHTTPS
+)
+
+// RequestTransportType extracts the transport type of the HTTP request. If
+// the transport cannot be found in the context, it returns the zero value
+// TransportTypeUnknown.
+func RequestTransportType(r *http.Request) TransportType {
+	if r == nil {
+		return TransportTypeUnknown
+	}
+	// If the assertion fails here, transport will always be the zero
+	// value of TransportType (TransportTypeUnknown).
+	transport, _ := r.Context().Value(TransportTypeKey{}).(TransportType)
+	return transport
+}
+
+// String returns a string representation of the transport type.
+func (t TransportType) String() string {
+	switch t {
+	case TransportTypeUnixSocket:
+		return "local"
+	case TransportTypeHTTP:
+		return "HTTP"
+	case TransportTypeHTTPS:
+		return "HTTPS"
+	default:
+		return "unknown"
+	}
+}
+
+// IsSupported returns true of the transport type is not unknown.
+func (t TransportType) IsSupported() bool {
```

Comment:
What about calling this IsValid?

```suggestion
// IsValid reports whether the transport type is valid.
func (t TransportType) IsValid() bool {
```

Reply:
Fixed, thank you. Resolving.